greenheart/simulation/greenheart_simulation.py

```diff
@@ -234,6 +285,61 @@ class GreenHeartSimulationOutput:
 
     platform_results: dict | None = field(default=None)
 
+    def save_to_file(self, filename: str):
+        """Saves select attributes of the class to a YAML file."""
+
+        # Convert the object to a dictionary of serializable types
+        serialized_data = {}
+        for attr in dir(self):
+            # Avoid private attributes and methods
+            if attr.startswith("_") or callable(getattr(self, attr)):
+                continue
+            if attr == "greenheart_config":  # fails: max recursion depth
+                continue
+            if attr == "hopp_interface":  # fails: max recursion depth
+                continue
+            if attr.startswith("profast"):  # fails: cannot pickle `dict_keys` object
+                continue
+            if attr == "hopp_results":  # fails: max recursion depth
+                continue
+            try:
+                value = getattr(self, attr)
+                serialized_data[attr] = convert_to_serializable(value)
+            except AttributeError:
+                pass
+
+        with Path.open(filename, "w") as file:
+            yaml.safe_dump(
+                serialized_data, file, default_flow_style=False, allow_unicode=True, sort_keys=False
+            )
+
+
+def load_greenheart_simulation_output_from_file(cls, filename: str):
+    """Loads the class attributes from a YAML file."""
```

Comment:
Could you please expand out this docstring with args and returns?

Reply:
Done